src/ar_metrics.erl

```diff
@@ -326,20 +331,24 @@ duration(StartKey, EndKey, Timings) ->
   End - Start.
 
 size(Sizes, response) ->
-  size(Sizes, response_headers) +
-    size(Sizes, response_body);
+	size_safe_add(size(Sizes, response_headers), size(Sizes, response_body));
 size(Sizes, response_headers) ->
-  proplists:get_value(resp_headers, Sizes);
+	proplists:get_value(resp_headers, Sizes);
 size(Sizes, response_body) ->
-  case proplists:get_value(chunks, Sizes) of
-    undefined ->
-      case proplists:get_value(file, Sizes) of
-        undefined ->
-          proplists:get_value(resp_body, Sizes);
-        FileSize -> FileSize
-      end;
-    ChunksSize -> ChunksSize
-  end.
+	case proplists:get_value(chunks, Sizes) of
+		undefined ->
+			case proplists:get_value(file, Sizes) of
+				undefined ->
+					proplists:get_value(resp_body, Sizes);
+				FileSize -> FileSize
+			end;
+		ChunksSize -> ChunksSize
+	end.
+
+size_safe_add(undefined, undefined) -> 0;
```

Comment:
Isn't it more a `safe_size_add`?  Also would like comments here and on other functions. Yeah, I know, I'm one of those bad comment-all-and-everything-guys. 😁